src/plugins/vis_builder/public/application/utils/use/use_persisted_agg_params.test.tsx

```diff
@@ -0,0 +1,191 @@
+    oldAggParam.schema = 'old metric';
+    aggMap.set('old metric', {
+      currentCount: 1,
+      maxCount: 1,
+      name: 'new metric',
+    });
+    const newAgg = updateAggParams(oldAggParam, aggMap);
+    expect(newAgg).toEqual({
+      type: '',
+      enabled: true,
+      schema: undefined,
+    });
+  });
+});
+
+describe('getSchemaMapping', () => {
+  let oldVisualizationType: Schema[];
+  let newVisualizationType: Schema[];
+  let schemaMetricTemplate: Schema;
+  let schemaBucketTemplate: Schema;
+
+  beforeEach(() => {
+    schemaMetricTemplate = {
+      aggFilter: [],
+      editor: '',
+      group: AggGroupNames.Metrics,
+      max: 1,
+      min: 1,
+      name: '',
+      params: [],
+      title: '',
+      defaults: '',
+    };
+    schemaBucketTemplate = {
+      aggFilter: [],
+      editor: '',
+      group: AggGroupNames.Buckets,
+      max: 1,
+      min: 1,
+      name: '',
+      params: [],
+      title: '',
+      defaults: '',
+    };
+  });
+
+  test('map metric schema to metric schema, bucket schema to bucket schema according to order', async () => {
+    oldVisualizationType = [
+      { ...schemaMetricTemplate, name: 'old metric 1' },
+      { ...schemaBucketTemplate, name: 'old bucket 1' },
+      { ...schemaMetricTemplate, name: 'old metric 2' },
+      { ...schemaBucketTemplate, name: 'old bucket 2' },
+      { ...schemaBucketTemplate, name: 'old bucket 3' },
+      { ...schemaBucketTemplate, name: 'old bucket 4' },
+      { ...schemaBucketTemplate, group: AggGroupNames.None, name: 'none' },
+    ];
+    newVisualizationType = [
+      { ...schemaMetricTemplate, name: 'new metric 1', max: 1 },
+      { ...schemaMetricTemplate, name: 'new metric 2', max: 2 },
+      { ...schemaMetricTemplate, name: 'new metric 3', max: 3 },
+      { ...schemaBucketTemplate, name: 'new bucket 1', max: 4 },
+      { ...schemaBucketTemplate, name: 'new bucket 2', max: 5 },
+      { ...schemaBucketTemplate, name: 'new bucket 3', max: 6 },
+    ];
+    const mappingResult = getSchemaMapping(oldVisualizationType, newVisualizationType);
+    expect(mappingResult).toMatchInlineSnapshot(`
+      Map {
+        "old metric 1" => Object {
+          "currentCount": 0,
+          "maxCount": 1,
+          "name": "new metric 1",
+        },
+        "old metric 2" => Object {
+          "currentCount": 0,
+          "maxCount": 2,
+          "name": "new metric 2",
+        },
+        "old bucket 1" => Object {
+          "currentCount": 0,
+          "maxCount": 4,
+          "name": "new bucket 1",
+        },
+        "old bucket 2" => Object {
+          "currentCount": 0,
+          "maxCount": 5,
+          "name": "new bucket 2",
+        },
+        "old bucket 3" => Object {
+          "currentCount": 0,
+          "maxCount": 6,
+          "name": "new bucket 3",
+        },
+      }
+    `);
+  });
+});
+
+describe('usePersistedAggParams', () => {
```

Comment:
Will rewrite the unit tests to test usePersistedAggParams() as a whole. I thought since i broke the function down to couple different helper functions so it is okay to just write unit tests for each individual helper function.